idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py

```diff
@@ -0,0 +1,380 @@
+        def pureCO2_temp_eqn(b, t):
+            return b.inlet_temperature[t] == b.pureCO2_temperature[t]
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Temperature")
+        def exh_temp_eqn(b, t):
+            return b.inlet_temperature[t] == b.exhaust_gas_temperature[t]
+
+        # surrogates and additional constraints/expressions
+        ''' This section is to add the surrogate models'''
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Specific reboiler duty in GJ/t CO2 or MJ/ kg CO2")
+        def SRD_eqn(b, t):
+            x1 = b.lean_loading[t]
+            x2 = b.Pz_mol[t]
+            return b.SRD[t] == (- 15565.227507172588957474 * x1
+                                + 8.9237367058697429911263 * log(x1)
+                                + 15285.362653727905126289 * exp(x1)
+                                - 6741.0217492965430210461 * x1**2
+                                - 3865.5900513667870654899 * x1**3
+                                + 0.17312045830047528404555E-002 * x2**3
+                                - 1.2960483459315448317994 * x1*x2
+                                - 15235.956076189686427824)
+
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Reboiler duty in MW")
+        def reboiler_duty(b, t):   # mol/s /1000 = kgmol/s * 44 kg/kgmol = kg / s
+            return b.SRD[t] * (b.pureCO2_flow_mol_comp[t, "CO2"]
+                               * 44.01 / 1000)
+
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Lean loading flowrate - kmol.hr")
+        def LL_flowrate(b, t):
+            x1 = b.lean_loading[t]
+            x2 = b.Pz_mol[t]
+            return (- 4402167.2503122268244624 * x1
+                    - 4.4210697568168892956919 * x2
+                    + 42394.191672889020992443 * log(x1)
+                    + 4021949.9049337119795382 * exp(x1)
+                    - 1394566.6646474981680512 * x1**2
+                    - 1225122.9083037786185741 * x1**3
+                    - 3898754.8073137737810612)
+
+
+    def initialize(blk,
+                   outlvl=idaeslog.NOTSET,
+                   solver='ipopt',
+                   optarg={'tol': 1e-6}):
+        '''
+        CO2 pure pyomo block initialization routine
+
+        Keyword Arguments:
+            outlvl : sets output level of initialisation routine
+
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        '''
+        iscale.calculate_scaling_factors(blk)  # remove to solve using baron
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        opt = pyo.SolverFactory(solver)
```

Comment:
will it be better to use the '`get_solver`' method here? that will also change the args in `initialize `method `solver=None, optarg=None`